### ISSUE TITLE
First run of background_job gives exception unable to open shape_predictor_5_face_landmarks.dat for reading

### ISSUE THREAD

**Issue by BurningTheSky** [matiasdelellis/facerecognition#481]
### Expected behaviour

background_job would analyse files for faces

### Actual behaviour

DlibHogModel.php shows an exception at line 193:  Unable to open /mnt/ncdata//appdata_oc9k17bhaf4u/facerecognition/models//3/shape_predictor_5_face_landmarks.dat for reading.

### Steps to reproduce
Run sudo -u www-data php /var/www/nextcloud/occ -v face:background_job -u burningsky

### Server configuration

* **Operating system:**  Ubuntu 20.04.2 LTS

* **Pdlib version:** 1.0.2

* **How is DLib installed:** Not sure, but the test runs correctly

* **How is PDlib installed:** sudo echo "deb https://repo.delellis.com.ar focal focal" > /etc/apt/sources.list.d/20-pdlib.list

* **PHP version:** 7.4.3

* **Web server:**  Apache/2.4.41

* **Database:** MariaDB 10.4.19

* **Nextcloud version:** 21.0.1

#### Background task log with debug.
<details>
<summary>sudo -u apache php occ -vvv face:background_job</summary>

```
burningsky@nextcloud:~/pdlib-min-test-suite$ sudo -u www-data php /var/www/nextcloud/occ -v face:background_job -u burningsky
1/10 - Executing task CheckRequirementsTask (Check all requirements)
        System: Linux
        System memory: 8344420352
        PHP Memory Limit: Unknown
2/10 - Executing task CheckCronTask (Check that service is started from either cron or from command)
3/10 - Executing task LockTask (Acquire lock so that only one background task can run)
4/10 - Executing task DisabledUserRemovalTask (Purge all the information of a user when disable the analysis.)
yielding
5/10 - Executing task StaleImagesRemovalTask (Crawl for stale images (either missing in filesystem or under .nomedia) and remove them from DB)
        Skipping stale images removal for user burningsky as there is no need for it
6/10 - Executing task CreateClustersTask (Create new persons or update existing persons)
        Skipping cluster creation, not enough data (yet) collected. For cluster creation, you need either one of the following:
        * have 1000 faces already processed
        * or you need to have 95% of you images processed
        Use stats command to track progress
yielding
7/10 - Executing task AddMissingImagesTask (Crawl for missing images for each user and insert them in DB)
        Skipping full image scan for user burningsky
8/10 - Executing task EnumerateImagesMissingFacesTask (Find all images which don't have faces generated for them)
yielding
9/10 - Executing task ImageProcessingTask (Process all images to extract faces)
        NOTE: Starting face recognition. If you experience random crashes after this point, please look FAQ at https://github.com/matiasdelellis/facerecognition/wiki/FAQ
Error during background task execution
If error is not transient, this means that core component of face recognition is not working properly
and that quantity and quality of detected faces and person will be low or suboptimal.
You probably want to file an issue (please include exception below) to: https://github.com/matiasdelellis/facerecognition/issues

In DlibHogModel.php line 193:

  [Exception]
  Unable to open /mnt/ncdata//appdata_oc9k17bhaf4u/facerecognition/models//3/shape_predictor_5_face_landmarks.dat for reading.


Exception trace:
  at /var/www/nextcloud/apps/facerecognition/lib/Model/DlibHogModel/DlibHogModel.php:193
 FaceLandmarkDetection->__construct() at /var/www/nextcloud/apps/facerecognition/lib/Model/DlibHogModel/DlibHogModel.php:193
 OCA\FaceRecognition\Model\DlibHogModel\DlibHogModel->open() at /var/www/nextcloud/apps/facerecognition/lib/BackgroundJob/Tasks/ImageProcessingTask.php:113
 OCA\FaceRecognition\BackgroundJob\Tasks\ImageProcessingTask->execute() at /var/www/nextcloud/apps/facerecognition/lib/BackgroundJob/BackgroundService.php:136
 OCA\FaceRecognition\BackgroundJob\BackgroundService->execute() at /var/www/nextcloud/apps/facerecognition/lib/Command/BackgroundCommand.php:148
 OCA\FaceRecognition\Command\BackgroundCommand->execute() at /var/www/nextcloud/3rdparty/symfony/console/Command/Command.php:255
 Symfony\Component\Console\Command\Command->run() at /var/www/nextcloud/3rdparty/symfony/console/Application.php:1009
 Symfony\Component\Console\Application->doRunCommand() at /var/www/nextcloud/3rdparty/symfony/console/Application.php:273
 Symfony\Component\Console\Application->doRun() at /var/www/nextcloud/3rdparty/symfony/console/Application.php:149
 Symfony\Component\Console\Application->run() at /var/www/nextcloud/lib/private/Console/Application.php:215
 OC\Console\Application->run() at /var/www/nextcloud/console.php:100
 require_once() at /var/www/nextcloud/occ:11

face:background_job [-u|--user_id USER_ID] [-M|--max_image_area MAX_IMAGE_AREA] [--defer-clustering] [-t|--timeout TIMEOUT]
```
</details>

**Comment by matiasdelellis** [comment]
Hi @BurningTheSky 
Please check model files.:
`sudo -u www-data ls -la /mnt/ncdata//appdata_oc9k17bhaf4u/facerecognition/models//3/`

Remove model:
`sudo -u www-data /mnt/ncdata//appdata_oc9k17bhaf4u/facerecognition/models//3/`

.. and install again:
`sudo -u www-data face:setup -m 3`

**Comment by BurningTheSky** [comment]
Ah, wasn't installed. Is 3 the best model to use?

**Comment by matiasdelellis** [comment]
> Ah, wasn't installed.  

Ok.. 😅 

> Is 3 the best model to use?

Maybe model 4, but also need more resources and time.

**Comment by BurningTheSky** [comment]
I acted without reading and removed 3 before installing 4 which was stupid. When I try and reinstall 3 it says " The file /mnt/ncdata//appdata_oc9k17bhaf4u/facerecognition/models//3/shape_predictor_5_face_landmarks.dat exists or is not writable" even though there isn't anything in the folder. I have tried removing and recreating the models folder but that doesn't help. How can I fix this?

``burningsky@nextcloud:~$ sudo -u www-data ls -la /mnt/ncdata//appdata_oc9k17bhaf4u/facerecognition/models
total 0
drwxr-xr-x 1 www-data www-data  0 May 24 07:26 .
drwxr-xr-x 1 www-data www-data 12 May 24 07:26 ..``
